### PR TITLE
fix(angular/button): add correct class if icon is not a direct descendant

### PR DESCRIPTION
**Pull request by mhaertwig**
The `sbb-icon-button` class was not added if the icon is not a direct descendant, e.g. because it's wrapped in a link. This PR fixes this bug.

### CODE REVIEW

File: src/angular/button/button.spec.ts [hunk 1]
```diff
@@ -684,6 +684,9 @@ class ButtonIconTest extends ButtonTestBase {}
       <a sbb-alt-button type="button"><sbb-icon svgIcon="example"></sbb-icon></a>
       <a sbb-secondary-button type="button"><sbb-icon svgIcon="example"></sbb-icon></a>
       <a sbb-ghost-button type="button"><sbb-icon svgIcon="example"></sbb-icon></a>
+      <button sbb-button type="button">
```
Comment (kyubisation) [inline src/angular/button/button.spec.ts:687]: Is this valid HTML?

Comment (mhaertwig) [reply src/angular/button/button.spec.ts]: Yes, but makes no sense. I removed it.

Comment (mhaertwig) [reply src/angular/button/button.spec.ts]: No, you're right. I didn't know this.
image

This bug appeared in our Tekton-UI which is why I created this PR. I close it and write a comment there.

Comment (kyubisation) [reply src/angular/button/button.spec.ts]: I was vaguely aware that there was some restriction and wanted to be sure.
Thanks for checking! 👍

**Pull request closed without merging**